### PR TITLE
Signatures and Signature verification implemented

### PR DESCRIPTION
**Pull request by TymoteuszSurynt**
Signatures and signature verification implemented

### CODE REVIEW

File: server/apolloassistant/verification.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    h = SHA.new()
+    h.update(message)
+    with open(f"{os.getcwd()}/.envs/private_key.pem", "r") as pem_file:
+        private_key = RSA.importKey(pem_file.read())
+    signer = PKCS1_PSS.new(private_key)
+    return signer.sign(h)
+
+
+def verify(signature, message):
+    h = SHA.new()
```
Comment (piotrszyma) [inline server/apolloassistant/verification.py]: Why duplicate?

File: server/apolloassistant/verification.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    h.update(message)
+    h = SHA.new()
+    h.update(message)
+    with open(f"{os.getcwd()}/.envs/public_key.pem", "r") as pem_file:
```
Comment (piotrszyma) [inline server/apolloassistant/verification.py]: Please don't open file on every call, cache it in memory.

File: server/apolloassistant/verification.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    }
+    for k,v in rsa_pair.items():
+        with open(f"{os.getcwd()}/.envs/{k}.pem","w") as pem_file:
+            pem_file.write(v.exportKey("pem"))
+
+
+def sign(message):
+    h = SHA.new()
+    h.update(message)
+    with open(f"{os.getcwd()}/.envs/private_key.pem", "r") as pem_file:
```
Comment (piotrszyma) [inline server/apolloassistant/verification.py]: Please create `settings.RSA_KEYS_PATH` and setup it relative to `APPS_DIR` and in here use `from django.conf import settings`.

**Comment by piotrszyma** [comment]
Please update poetry.lock! Probably using `docker-compose exec app poetry update`

**Review by rrothenberger — CHANGES_REQUESTED** [review]
Please fix failing CI build before merging.

File: server/config/settings/base.py [hunk 4]
```diff
@@ -262,3 +262,7 @@
 CORS_ORIGIN_ALLOW_ALL = True
 
 LOGIN_REDIRECT_URL = '/'
+
+REGISTER_KEYS_PATH = str(APPS_DIR.path(".envs"))
+PUBLIC_KEY = None
+PRIVATE_KEY = None
```
Comment (rrothenberger) [inline server/config/settings/base.py]: Add it to production config as well. And do not use paths out of server directory. They are not part of docker image.

**Review by rrothenberger — CHANGES_REQUESTED** [review]

**Review by rrothenberger — APPROVED** [review]

**Pull request merged**